tests/pkg/fatfs/Makefile

```diff
@@ -31,3 +21,13 @@ compressed-image:
 	${Q}./create_fat_image_file.sh $(FATFS_IMAGE_FILE_SIZE_MIB)
 
 include $(RIOTBASE)/Makefile.include
+
+ifeq ($(BOARD),native)
+  #overwrite default mtd_native-config to use fat image as flash device
+  CFLAGS += -DMTD_NATIVE_FILENAME=\"./bin/riot_fatfs_disk.img\"
+  CFLAGS += -DFATFS_IMAGE_FILE_SIZE_MIB=$(FATFS_IMAGE_FILE_SIZE_MIB)
+  CFLAGS += -DMTD_SECTOR_NUM=\(\(\(FATFS_IMAGE_FILE_SIZE_MIB\)*1024*1024\)/MTD_SECTOR_SIZE\)
+else ifeq (,$(filter mtd_sdmmc,$(USEMODULE)))
```

Comment:
Unfortunatly, it doesn't work in that way if `mtd_sdmmc` isn't used since the dependencies for `mtd_sdcard` aren't resolved :worried: Either we let these tests as they are and use `mtd_sdcard` even if the board has a `mtd_sdmmc` device or we need another solution.

Reply:
@benpicco I finally solved the problem by leaving the makefiles as they were before, even though the `mtd_sdcard` module is activated in addition to the `mtd_sdmmc` module when the board uses it. But I changed the `main.c` files to use the SD/MMC device `mtd_sdmmc_dev0` first if the module `mtd_sdmmc` is enabled, and the `SPI SD Card` only if the module `mtd_sdmmc` is not enabled, for example https://github.com/RIOT-OS/RIOT/blob/66acaf4ea45be288731a3fe150fc07ddaa17c18a/tests/pkg/fatfs/main.c#L413-L419